<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="map_file" value="$(find tsuten_navigation)/map/field.yaml" />
  <arg name="rviz_config" value="$(find tsuten_navigation)/rviz/main.rviz" />

  <include file="$(find tsuten_description)/launch/spawn_robot.launch" />

  <node name="map_server" pkg="map_server" type="map_server"
    args="$(arg map_file)" />

  <node name="fake_localization" pkg="fake_localization" type="fake_localization" output="screen">
    <remap from="base_pose_ground_truth" to="odom" />
  </node>

  <node name="move_base" pkg="move_base" type="move_base" output="screen" clear_params="true">
    <rosparam file="$(find tsuten_navigation)/config/move_base/move_base_params.yaml"
      command="load" />
    <rosparam file="$(find tsuten_navigation)/config/move_base/costmap_common_params.yaml"
      command="load" ns="global_costmap" />
    <rosparam file="$(find tsuten_navigation)/config/move_base/costmap_common_params.yaml"
      command="load" ns="local_costmap" />
    <rosparam file="$(find tsuten_navigation)/config/move_base/global_costmap_params.yaml"
      command="load" />
    <rosparam file="$(find tsuten_navigation)/config/move_base/local_costmap_params.yaml"
      command="load" />
    <rosparam file="$(find tsuten_navigation)/config/move_base/base_local_planner_params.yaml"
      command="load" />
  </node>

  <node name="four_omni_drive_controller" pkg="tsuten_navigation"
    type="four_omni_drive_controller">
    <param name="wheel_radius" value="30.6e-3" />
    <param name="wheel_vels_topic" value="wheel_joint_velocity_controller/command" />
  </node>

  <node name="odom_tf_publisher" pkg="tsuten_navigation" type="odom_tf_publisher" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rviz_config)" required="true" />

  <!-- <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"
    launch-prefix="bash -c 'sleep 3.0; $0 $@' " /> -->
</launch>